File: 14/longestCommonPrefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        s = ""
        if len(strs) == 0:
            return s
        min_length = 9999999999999999
        for i in strs:
            if i == "":
                return s
            if min_length > len(i):
                min_length = len(i)
        index = 0
        while index < min_length:
            tmp = strs[0][index]
            for i in strs:
                if i[index] != tmp:
                    return s
            s += tmp
            index += 1
        return s

File: 14/longestCommonPrefix.cpp
#include<iostream>
#include<vector>
#include<string>
using namespace std;

string longestCommonPrefix(vector<string>& strs)
{
	if(strs.size() == 0)
		return "";
	int count = 0;
	string s = strs[0];
	while(true)
	{
		for(int i = 1; i < strs.size(); i++)
		{
			if(count == strs[i].size())
				return strs[0].substr(0,count - 1);
			else
			{
				if (strs[i][count] != s[count])
					return strs[0].substr(0, count - 1);
			}
		}
		count ++;
	}
	
}

int main()
{
	vector<string> a = {"123","1234","12345"};
	cout<<longestCommonPrefix(a);
}